src/core/ActivityInfoTree.h

```diff
@@ -59,12 +61,17 @@ protected slots:
 private:
     // this is the full activity list, it never changes
     QList<ActivityInfo *> m_menuTreeFull;
+    QList<ActivityInfo *> m_menuTreeFulltemp;
     // represents the Menu view and can be filtered
     QList<ActivityInfo *> m_menuTree;
+    QList<ActivityInfo *> m_searched;
     ActivityInfo *m_rootMenu;
     ActivityInfo *m_currentActivity;
     static int menuTreeCount(QQmlListProperty<ActivityInfo> *property);
     static ActivityInfo *menuTreeAt(QQmlListProperty<ActivityInfo> *property, int index);
+    QObject *m_mainWindow;
+    int templength = 0;
+    QString searchedtext;
```

Comment:
Premature optimization is the root of all evil :); If the issue is in micro seconds and not at least in milliseconds, I don't think it's useful.
There are other "optimizations" that can be done without removing code readability: using const QString & for the parameter, using std algorithms: erase, remove_if.
The loop is done for in the worst case 100 times (the number of activities) and only on user inputs. It's not a lot and probably not the worst part of time used. As you said, for the 'u', it took more time than for the 't'. So I agree with you that less loops are better in theory but in this case, I think it's not useful to lose in code readability to earn a few micro seconds that won't be visible by the user.